src/main/java/org/jenkinsci/plugins/mesos/JenkinsScheduler.java

```diff
@@ -556,6 +571,42 @@ private void createMesosTask(Offer offer, Request request) {
 
     results.put(taskId, new Result(request.result, new Mesos.JenkinsSlave(offer.getSlaveId()
         .getValue())));
+    finishedTasks.add(taskId);
+  }
+
+  /**
+   * Checks if the given taskId already exists or just finished running. If it has, then refuse the offer.
+   * @param offer The offer to check if the task is already running
+   * @param taskId The task id
+   * @return True if the task already exists, false otherwise
+   */
+  @VisibleForTesting
```

Comment:
why this annotation here and below? are you planning to add tests? it would be great, if you plan to by the way!